### ISSUE TITLE
ESLint fail in Jenkins and local dev, but passing on Travis

### ISSUE THREAD

**Issue by Krinkle** [qunitjs/qunit#1248]
The "test" npm run-script on Travis CI is passing in Node 4, 8 and 9 at <https://travis-ci.org/qunitjs/qunit/builds/326175125>.

Node 8, for example at <https://travis-ci.org/qunitjs/qunit/jobs/326175130> passes:

```
$ node --version
v8.9.4
$ npm --version
5.6.0
..
$ yarn ..
$ export NPM_SCRIPT=test
$ npm run-script $NPM_SCRIPT
  > grunt  ..
  Running "eslint:js" (eslint) task

  ..
  Done.
```

Yet, the same command is failing on Jenkins at <http://jenkins.jquery.com/job/QUnit/746/console>

```
+ node --version
v8.9.3
+ npm --version
5.5.1
+ npm install ..
+ grunt ..
Running "eslint:js" (eslint) task

/var/lib/jenkins/jobs/QUnit/workspace/test/main/deepEqual.js
  1808:43  error  Multiple spaces found before '0'    no-multi-spaces
  1877:8   error  Multiple spaces found before 'new'  no-multi-spaces

✖ 2 problems (2 errors, 0 warnings)
  2 errors, 0 warnings potentially fixable with the `--fix` option.

Warning: Task "eslint:js" failed.
```

I can reproduce the same failure locally as well:

```
$ node --version
v9.3.0
$ npm --version
5.6.0
$ npm install ..
$ npm test ..
> grunt ..
Running "copy:src-css" (copy) task
Copied 1 file

Running "eslint:js" (eslint) task

/Users/krinkle/Development/qunit/test/main/deepEqual.js
  1808:43  error  Multiple spaces found before '0'    no-multi-spaces
  1877:8   error  Multiple spaces found before 'new'  no-multi-spaces

✖ 2 problems (2 errors, 0 warnings)
  2 errors, 0 warnings potentially fixable with the `--fix` option.

Warning: Task "eslint:js" failed.
```

**Comment by Krinkle** [comment]
These appear to be genuine failures. The question is, why is the Travis build passing?

**Comment by rwjblue** [comment]
Travis config is using `yarn` (locked dependencies) but Jenkins is using `npm install` (without a lockfile).  I believe to make Travis fail in the same way, you would need to `yarn upgrade`...

**Comment by Krinkle** [comment]
@rwjblue Good catch, thanks. Yeah, that must be it.

**Comment by trentmwillis** [comment]
Fixed by #1249. [Passing Jenkins job](http://jenkins.jquery.com/job/QUnit/747/).
